x-pack/filebeat/input/http_endpoint/handler.go

```diff
@@ -124,7 +128,8 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			}
 		}
 
-		if err = h.publishEvent(obj, headers); err != nil {
+		acker.Add()
```

Comment:
With end-to-end acknowledgements the goal is to be able to provide a delivery guarantee that the event(s) was written to the destination output. So I think this needs to relay the feedback to the caller that the event was written to the output.

I envision this meaning that a request blocks until the input receives all of the ACKs that it expects. This leads to a few follow-up behavioral questions:

1. Introducing blocking behavior could be a viewed as a breaking change. Perhaps the waiting behavior should be opt-in and have a timeout like via [?wait_for_completion_timeout=30](https://github.com/elastic/elasticsearch-specification/blob/53edb7836cd4d15fc15c1341a5eb7bf9dfb5d50a/specification/_json_spec/async_search.get.json#L27-L29).
2. Should there be any "circuit breakers" and how should the API respond when it overloaded (e.g. receiving more requests or more events than it can process in a timely manner)? We want to avoid a DoS situation if we have some high volume clients and an output that is slow or down. (_This one is relevant even without E2E ack'ing._)

Reply:
That sounds good to me.